src/OpalCompiler-Core/OCInstanceScope.class.st

```diff
@@ -10,6 +10,11 @@ Class {
 	#category : #'OpalCompiler-Core-Semantics'
 }
 
+{ #category : #'instance creation' }
+OCInstanceScope class >> for: aClass [
+	^ self new slots: aClass allSlots; outerScope: aClass; yourself
```

Comment:
I you going to get rid of OCInstanceScope as well?

Reply:
Yes! That is the next step: we do not need to add all slots to a scope, we can just look the up in #lookupVar: of the class.

Then the class is the scope of a Slot (this is, we can implement #scope there, too.. nice for uniformity: every Variable has a scope, lookupVar: withe the name of the var will return that variable).

Reply:
this then shows that self, super and thisContext are variables of the environment scope... there is one instance of SelfVariable per SystemDictionary instance.

Reply:
Indeed, It looks very nice